File: apps/web/src/types/evaluation.ts
/**
 * API contract for an ephemeral evaluation run. Preview results are not
 * persisted, which lets the playground evaluate ad-hoc seller responses.
 */
export interface EvaluationPreviewRequest {
  prospect_context?: string;
  conversation_context?: string;
  objection: string;
  seller_response: string;
  sales_stage: string;
  industry?: string;
  company_context?: string;
}

export interface EvaluationDimensionScore {
  dimension: string;
  score: number;
  reasoning: string;
  /** The API may return one excerpt or a collection of supporting excerpts. */
  evidence: string | string[];
}

export type ConfidenceLevel = "low" | "medium" | "high" | (string & {});

export interface EvaluationPreviewResponse {
  overall_score: number;
  response_quality_score: number;
  objection_handling_score: number;
  forward_momentum_score: number;
  dimensions: EvaluationDimensionScore[];
  strengths: string[];
  weaknesses: string[];
  missed_opportunities: string[];
  recommended_strategy: string;
  confidence_score: number;
  confidence_level: ConfidenceLevel;
  uncertainty_reasons: string[];
}

File: services/api/app/schemas/evaluation.py
from pydantic import BaseModel, ConfigDict
from typing import List, Dict, Any, Optional
import uuid
from datetime import datetime

class EvaluationCreate(BaseModel):
    scenario_id: uuid.UUID
    seller_response: str

class EvaluationPreviewRequest(BaseModel):
    prospect_context: Optional[str] = None
    conversation_context: Optional[str] = None
    objection: str
    seller_response: str
    sales_stage: str
    industry: Optional[str] = None
    company_context: Optional[str] = None

class EvaluationPreviewResponse(BaseModel):
    overall_score: float
    response_quality_score: float
    objection_handling_score: float
    forward_momentum_score: float
    evaluation_version: str
    strengths: List[str]
    weaknesses: List[str]
    missed_opportunities: List[str]
    recommended_strategy: str
    dimensions: List[Dict[str, Any]]

class DimensionScoreResponse(BaseModel):
    id: uuid.UUID
    dimension: str
    score: float
    reasoning: str
    evidence: str

    model_config = ConfigDict(from_attributes=True)

class EvaluationResponse(BaseModel):
    id: uuid.UUID
    scenario_id: uuid.UUID
    seller_response: str
    overall_score: float
    response_quality_score: float
    objection_handling_score: float
    forward_momentum_score: float
    evaluation_version: str
    strengths: List[str]
    weaknesses: List[str]
    missed_opportunities: List[str]
    recommended_strategy: str
    created_at: datetime
    scores: List[DimensionScoreResponse]

    model_config = ConfigDict(from_attributes=True)
